torch_geometric/edge_index.py

```diff
@@ -1364,239 +1408,209 @@ def flip(
+def _add(
     input: EdgeIndex,
-    other: Union[int, Tensor],
-    out: Tensor,
+    other: Union[int, Tensor, EdgeIndex],
+    *,
     alpha: int = 1,
 ) -> Union[EdgeIndex, Tensor]:
 
+    out = aten.add.Tensor(
+        input._data,
+        other._data if isinstance(other, EdgeIndex) else other,
+        alpha=alpha,
+    )
+
     if out.dtype not in SUPPORTED_DTYPES:
         return out
     if out.dim() != 2 or out.size(0) != 2:
         return out
 
-    output: EdgeIndex = out.as_subclass(EdgeIndex)
+    out = EdgeIndex(out)
+
+    if isinstance(other, Tensor) and other.numel() <= 1:
+        other = int(other)
 
     if isinstance(other, int):
         size = maybe_add(input._sparse_size, other, alpha)
         assert len(size) == 2
-        output._sparse_size = size
-        output._sort_order = input._sort_order
-        output._is_undirected = input.is_undirected
-        output._T_perm = input._T_perm
-
-    elif isinstance(other, Tensor) and other.numel() <= 1:
-        size = maybe_add(input._sparse_size, int(other), alpha)
-        assert len(size) == 2
-        output._sparse_size = size
-        output._sort_order = input._sort_order
-        output._is_undirected = input.is_undirected
-        output._T_perm = input._T_perm
+        out._sparse_size = size
+        out._sort_order = input._sort_order
+        out._is_undirected = input.is_undirected
+        out._T_perm = input._T_perm
 
     elif isinstance(other, Tensor) and other.size() == (2, 1):
         size = maybe_add(input._sparse_size, other.view(-1).tolist(), alpha)
         assert len(size) == 2
-        output._sparse_size = size
-        output._sort_order = input._sort_order
-        output._T_perm = input._T_perm
+        out._sparse_size = size
+        out._sort_order = input._sort_order
         if torch.equal(other[0], other[1]):
-            output._is_undirected = input.is_undirected
+            out._is_undirected = input.is_undirected
+        out._T_perm = input._T_perm
 
     elif isinstance(other, EdgeIndex):
         size = maybe_add(input._sparse_size, other._sparse_size, alpha)
         assert len(size) == 2
-        output._sparse_size = size
-
-    return output
-
+        out._sparse_size = size
 
-@implements(torch.add)
-@implements(Tensor.add)
-def add(
-    input: EdgeIndex,
-    other: Union[int, Tensor],
-    *,
-    alpha: int = 1,
-    out: Optional[Tensor] = None,
-) -> Union[EdgeIndex, Tensor]:
+    return out
 
-    output = Tensor.__torch_function__(  #
-        torch.add, (Tensor, ), (input, other), dict(alpha=alpha, out=out))
 
-    return postprocess_add_(input, other, output, alpha)
+# @implements(aten.add_.Tensor)
+# def add_(
+#     input: EdgeIndex,
+#     other: Union[int, Tensor, EdgeIndex],
+#     *,
+#     alpha: int = 1,
+# ) -> Tensor:
+#     aten.add_.Tensor(
+#         input._data,
+#         other._data if isinstance(other, EdgeIndex) else other,
+#         alpha=alpha,
+#     )
+#     return input
```

Comment:
could be nit: it this still relevant?

Reply:
commented out on purpose, and brought back in #9239